stixcore/idb/idb.py

```diff
@@ -572,3 +995,142 @@ def get_idb_version(self):
+                FROM PID , PLF, PCF, tblConfigS2KParameterTypes as PTYPE
+                WHERE
+                    PLF.PLF_NAME = PCF.PCF_NAME
+                    AND PID_TYPE = ?
+                    AND PID_STYPE = ?
+                    AND PLF.PLF_SPID = PID.PID_SPID
+                    AND PTYPE.PTC = PCF.PCF_PTC
+                    AND PCF.PCF_PFC >= PTYPE.PFC_LB
+                    AND PTYPE.PFC_UB >= PCF.PCF_PFC
+                ORDER BY
+                    PLF.PLF_OFFBY asc ''')
+        parameters = self._execute(sql, (service_type, service_subtype), 'dict')
+
+        parent = IdbPacketTree()
+        for par in parameters:
+            parObj = IdbStaticParameter(par)
+            node = self._create_parse_node(parObj.PCF_NAME, parObj, 0, [])
+            parent.children.append(node)
+        return parent
+
+    def _create_parse_node(self, name, parameter=None, counter=0, children=None):
+
+        if children is None:
+            children = []
+
+        parameter.bin_format = self._get_stream_type_format(parameter.S2K_TYPE, parameter.PCF_WIDTH)
+        node = IdbPacketTree(name=name, counter=counter, parameter=parameter, children=children)
+        return node
+
+    def get_variable_structure(self, service_type, service_subtype, ssid=None):
+        """Create a dynamic parse tree for the specified TM packet.
+
+        Parameters
+        ----------
+        service_type : `int`
+            The TM packet service type.
+        service_subtype : `int`
+            The TM packet service subtype.
+        ssid : `int` optional
+            The TM packet SSID. Default to `None`
+
+        Returns
+        -------
+        `~stixcore/idb/idb/IdbPacketTree`
+            The IdbPacketTree implements nested repeaters.
+        """
+
+        sql = (f'''SELECT
+                    PID_SPID, PID_DESCR, PID_TPSD,
+                    PCF.PCF_NAME, VPD.VPD_POS,PCF.PCF_WIDTH,PCF.PCF_PFC, PCF.PCF_PTC,VPD.VPD_OFFSET,
+                    VPD.VPD_GRPSIZE,PCF.PCF_DESCR ,PCF.PCF_CURTX,
+                    S2K_TYPE
+                FROM PID , VPD, PCF, tblConfigS2KParameterTypes as PTYPE
+                WHERE
+                    PID_TYPE = ?
+                    AND PID_STYPE = ?
+                    {"AND PID_PI1_VAL = ? " if ssid is not None else " "}
+                    AND VPD.VPD_NAME = PCF.PCF_NAME
+                    AND VPD.VPD_TPSD = PID.PID_SPID
+                    AND PTYPE.PTC = PCF.PCF_PTC
+                    AND PCF.PCF_PFC >= PTYPE.PFC_LB
+                    AND PTYPE.PFC_UB >= PCF.PCF_PFC
+                ORDER BY
+                    VPD.VPD_POS asc ''')
+        args = (service_type, service_subtype)
+        if ssid is not None:
+            args = args + (ssid,)
+        param_pcf_structures = self._execute(sql, args, 'dict')
+
+        repeater = [{'node': IdbPacketTree(), 'counter': 1024}]
+
+        for par in param_pcf_structures:
+            parObj = IdbVariableParameter(par)
+            if repeater:
```

Comment:
with `repeater` defined as above is this `if` not always true?

Reply:
repeater.pop() can empty the repeater list and so this statement will become false